### PR TITLE
PLANNER-2715 Variable listener cleanup

### PR DESCRIPTION
**Pull request by yurloc**
### JIRA
https://issues.redhat.com/browse/PLANNER-2715

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
  
- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: 
    please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: 
    add the label `run_fdb`

- for a <b>compile downstream build</b>  
  please add comment: <b>Jenkins run cdb</b>

- for a <b>full production downstream build</b>  
  please add comment: <b>Jenkins execute product fdb</b>

- for an <b>upstream build</b>  
  please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] mandrel</b>
</details>

### CI Status

 You can check OptaPlanner repositories CI status from [Chain Status webpage](https://kiegroup.github.io/optaplanner/).

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - optaplanner-quickstarts job** `#66` was: **NOT_BUILT**
Possible explanation: Woops ... I don't know about this result value ... Please ask maintainer.

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/66/display/redirect or see console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /tmp/jenkins-gitclient-ssh2929373820958382606.key<br/> > git ls-remote -h https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Found match: refs/heads/main revision a93f11abbb2b36b09dd1298ff7daa104c00d6c6c<br/>The recommended git tool is: NONE<br/>using credential kie-ci-user-key<br/> > git rev-parse --resolve-git-dir /var/lib/jenkins/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/optaplanner.tests.downstream.optaplanner-quickstarts/workspace@libs/d03f8b53ef89e88ec2b15428a250a4b1fb7910bfcdd3e1c33e4e95c3adb758bb/.git # timeout=10<br/>Fetching changes from the remote Git repository<br/> > git config remote.origin.url https://github.com/kiegroup/jenkins-pipeline-shared-libraries # timeout=10<br/>Fetching without tags<br/>Fetching upstream changes from https://github.com/kiegroup/jenkins-pipeline-shared-libraries<br/> > git --version # timeout=10<br/> > git --version # 'git version 1.8.3.1'<br/>using GIT_SSH to set credentials SSH: kie-ci service account with access to jboss-integration private github space<br/>[INFO] Currently running in a labeled security context<br/> > /usr/bin/chcon --type=ssh_home_t /var/lib/jenkins/jobs/KIE/jobs/kogito/jobs/main/jobs/pullrequest/jobs/optaplanner.tests.downstream.optaplanner-quickstarts/workspace@libs/d03f8b53ef89e88ec2b15428a250a4b1fb7910bfcdd3e1c33e4e95c3adb758bb@tmp/jenkins-gitclient-ssh3053071169711674522.key<br/> > git fetch --no-tags --progress https://github.com/kiegroup/jenkins-pipeline-shared-libraries +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Checking out Revision a93f11abbb2b36b09dd1298ff7daa104c00d6c6c (main)<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f a93f11abbb2b36b09dd1298ff7daa104c00d6c6c # timeout=10<br/>Commit message: "Extracted the Maven Settings file (#166)"<br/> > git rev-list --no-walk a93f11abbb2b36b09dd1298ff7daa104c00d6c6c # timeout=10<br/>[Pipeline] Start of Pipeline<br/>[Pipeline] node<br/>Still waiting to schedule task<br/>‘rhos-d-kie-rhel7-xlarge-721’ is offline<br/>Running on rhos-01-kie-psi-rhel7-xlarge-5 in /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Checkout SCM)<br/>[Pipeline] checkout<br/>The recommended git tool is: NONE<br/>using credential kie-ci<br/>Cloning the remote Git repository<br/>Cloning repository https://github.com/kiegroup/kogito-pipelines.git<br/> > git init /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts # timeout=10<br/>Fetching upstream changes from https://github.com/kiegroup/kogito-pipelines.git<br/> > git --version # timeout=10<br/> > git --version # 'git version 2.34.1'<br/>using GIT_ASKPASS to set credentials PSSWD: kie-ci service account with access to jboss-integration private github space<br/> > git fetch --tags --force --progress -- https://github.com/kiegroup/kogito-pipelines.git +refs/heads/*:refs/remotes/origin/* # timeout=10<br/>Avoid second fetch<br/>Checking out Revision 153096f389746b8c09c85297744ceaf2e36c6900 (origin/main)<br/> > git config remote.origin.url https://github.com/kiegroup/kogito-pipelines.git # timeout=10<br/> > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10<br/> > git rev-parse origin/main^{commit} # timeout=10<br/> > git config core.sparsecheckout # timeout=10<br/> > git checkout -f 153096f389746b8c09c85297744ceaf2e36c6900 # timeout=10<br/>Commit message: "prevented multiple starts of a PR & pipelines won't run on master (#456)"<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] timestamps<br/>[Pipeline] {<br/>[Pipeline] timeout<br/>[2022-05-27T12:55:30.786Z] Timeout set to expire in 3 hr 0 min<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Declarative: Tool Install)<br/>[Pipeline] tool<br/>[2022-05-27T12:55:30.889Z] Installing NodeJS from /var/lib/jenkins/caches/nodejs/LINUX/amd64/12.16.3.tar.gz to /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3 on rhos-01-kie-psi-rhel7-xlarge-5<br/>[Pipeline] envVarsForTool<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] stage<br/>[Pipeline] { (Initialize)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-05-27T12:55:33.043Z] + printenv<br/>[Pipeline] archiveArtifacts<br/>[2022-05-27T12:55:33.111Z] Archiving artifacts<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (check space before build)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] dir<br/>[2022-05-27T12:55:33.669Z] Running in /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2022-05-27T12:55:33.697Z] [INFO] space left on the machine<br/>[Pipeline] sh<br/>[2022-05-27T12:55:33.975Z] + df -h<br/>[2022-05-27T12:55:33.975Z] Filesystem      Size  Used Avail Use% Mounted on<br/>[2022-05-27T12:55:33.975Z] devtmpfs        7.8G     0  7.8G   0% /dev<br/>[2022-05-27T12:55:33.975Z] tmpfs           7.8G     0  7.8G   0% /dev/shm<br/>[2022-05-27T12:55:33.975Z] tmpfs           7.8G   17M  7.8G   1% /run<br/>[2022-05-27T12:55:33.975Z] tmpfs           7.8G     0  7.8G   0% /sys/fs/cgroup<br/>[2022-05-27T12:55:33.975Z] /dev/vda1        80G   18G   63G  23% /<br/>[2022-05-27T12:55:33.975Z] tmpfs           1.6G     0  1.6G   0% /run/user/1000<br/>[Pipeline] echo<br/>[2022-05-27T12:55:33.994Z] [INFO] space of /home/jenkins<br/>[Pipeline] sh<br/>[2022-05-27T12:55:34.272Z] + du -h -d1 /home/jenkins<br/>[2022-05-27T12:55:34.273Z] 4.0K	/home/jenkins/.ssh<br/>[2022-05-27T12:55:34.273Z] 0	/home/jenkins/go<br/>[2022-05-27T12:55:34.273Z] 8.0K	/home/jenkins/.vnc<br/>[2022-05-27T12:55:34.273Z] 8.0K	/home/jenkins/.m2<br/>[2022-05-27T12:55:34.273Z] 2.1G	/home/jenkins/git-repos<br/>[2022-05-27T12:55:34.273Z] 8.0K	/home/jenkins/.config<br/>[2022-05-27T12:55:34.273Z] 4.0K	/home/jenkins/.cekit<br/>[2022-05-27T12:55:34.273Z] 80M	/home/jenkins/virtenvs<br/>[2022-05-27T12:55:34.273Z] 30M	/home/jenkins/remoting<br/>[2022-05-27T12:55:34.273Z] 0	/home/jenkins/.cache<br/>[2022-05-27T12:55:34.273Z] 2.3M	/home/jenkins/workspace<br/>[2022-05-27T12:55:34.273Z] 0	/home/jenkins/.pki<br/>[2022-05-27T12:55:34.273Z] 81M	/home/jenkins/tools<br/>[2022-05-27T12:55:34.273Z] 0	/home/jenkins/caches<br/>[2022-05-27T12:55:34.273Z] 2.3G	/home/jenkins<br/>[Pipeline] echo<br/>[2022-05-27T12:55:34.290Z] [INFO] space of workspace<br/>[Pipeline] sh<br/>[2022-05-27T12:55:34.568Z] + du -h -d3 /home/jenkins/workspace<br/>[2022-05-27T12:55:34.569Z] 2.3M	/home/jenkins/workspace/KIE/kogito/main<br/>[2022-05-27T12:55:34.569Z] 2.3M	/home/jenkins/workspace/KIE/kogito<br/>[2022-05-27T12:55:34.569Z] 2.3M	/home/jenkins/workspace/KIE<br/>[2022-05-27T12:55:34.569Z] 2.3M	/home/jenkins/workspace<br/>[Pipeline] }<br/>[Pipeline] // dir<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Install build-chain tool)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2022-05-27T12:55:34.896Z] [INFO] Getting build-chain version from composite action file<br/>[Pipeline] readFile<br/>[Pipeline] echo<br/>[2022-05-27T12:55:35.024Z] [INFO] build-chain version recovered '2.3.34'<br/>[Pipeline] sh<br/>[2022-05-27T12:55:35.307Z] + npm install -g @kie/build-chain-action@2.3.34 -registry=https://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org/<br/>[2022-05-27T12:55:38.583Z] /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3/bin/build-chain-action -> /home/jenkins/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/nodejs-12.16.3/lib/node_modules/@kie/build-chain-action/bin/build-chain-cli.js<br/>[2022-05-27T12:55:38.583Z] npm WARN notsup Unsupported engine for @kie/build-chain-action@2.3.34: wanted: {"node":">= 12.18.0"} (current: {"node":"12.16.3","npm":"6.14.4"})<br/>[2022-05-27T12:55:38.583Z] npm WARN notsup Not compatible with your version of node/npm: @kie/build-chain-action@2.3.34<br/>[2022-05-27T12:55:38.583Z] npm WARN @octokit/plugin-request-log@1.0.4 requires a peer of @octokit/core@>=3 but none is installed. You must install peer dependencies yourself.<br/>[2022-05-27T12:55:38.583Z] <br/>[2022-05-27T12:55:38.583Z] + @kie/build-chain-action@2.3.34<br/>[2022-05-27T12:55:38.583Z] added 109 packages from 89 contributors in 2.622s<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] }<br/>[Pipeline] // withEnv<br/>[Pipeline] }<br/>[Pipeline] // stage<br/>[Pipeline] stage<br/>[Pipeline] { (Build projects)<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] tool<br/>[Pipeline] envVarsForTool<br/>[Pipeline] withEnv<br/>[Pipeline] {<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] echo<br/>[2022-05-27T12:55:39.283Z] BUILD_MVN_OPTS_CURRENT = -Dfull<br/>[Pipeline] configFileProvider<br/>[2022-05-27T12:55:39.296Z] provisioning config files...<br/>[2022-05-27T12:55:39.303Z] copy managed file [kogito_pr_settings] to file:/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts@tmp/config1513810815088492169tmp<br/>[Pipeline] {<br/>[Pipeline] withCredentials<br/>[2022-05-27T12:55:39.330Z] Masking supported pattern matches of $GITHUB_TOKEN<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-05-27T12:55:39.385Z] Warning: A secret was passed to "sh" using Groovy String interpolation, which is insecure.<br/>[2022-05-27T12:55:39.385Z] 		 Affected argument(s) used the following variable(s): [GITHUB_TOKEN]<br/>[2022-05-27T12:55:39.385Z] 		 See https://jenkins.io/redirect/groovy-string-interpolation for details.<br/>[2022-05-27T12:55:39.720Z] + build-chain-action -token=**** -df=https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml -folder=bc build pr -url=https://github.com/kiegroup/optaplanner/pull/2004 -sp=kiegroup/optaplanner-quickstarts --skipParallelCheckout -cct '(^mvn .*)||$1 -s /home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts@tmp/config1513810815088492169tmp -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.failure.ignore=true'<br/>[2022-05-27T12:55:40.027Z] [INFO]  Executing pull request flow for https://github.com/kiegroup/optaplanner/pull/2004 in bc<br/>[2022-05-27T12:55:40.292Z] [INFO]  https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml exists, using it<br/>[2022-05-27T12:55:40.292Z] ::group::[PRE] Executing pre section for https://raw.githubusercontent.com/kiegroup/kogito-pipelines/main/.ci/pull-request-config.yaml<br/>[2022-05-27T12:55:40.292Z] [INFO]  Execute command 'export BUILD_MVN_OPTS=" -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts'<br/>[2022-05-27T12:55:40.292Z] [INFO]  Treating export command since it's not possible to run it from the runner itself<br/>[2022-05-27T12:55:40.292Z] [INFO]  The variable BUILD_MVN_OPTS has been stored with ' -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3' value<br/>[2022-05-27T12:55:40.292Z] ::notice title= [PRE] [Execution OK]::export BUILD_MVN_OPTS=" -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"<br/>[2022-05-27T12:55:40.292Z] [INFO]  Execute command 'echo "BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts'<br/>[2022-05-27T12:55:40.292Z] [command]/usr/bin/echo BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3<br/>[2022-05-27T12:55:40.292Z] BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3<br/>[2022-05-27T12:55:40.292Z] ::notice title= [PRE] [Execution OK]::echo "BUILD_MVN_OPTS= -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false dependency:tree -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3"<br/>[2022-05-27T12:55:40.293Z] [INFO]  Execute command 'export BUILD_MVN_OPTS_CURRENT="-Dfull -Dquarkus.bootstrap.effective-model-builder"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts'<br/>[2022-05-27T12:55:40.293Z] [INFO]  Treating export command since it's not possible to run it from the runner itself<br/>[2022-05-27T12:55:40.293Z] [INFO]  The variable BUILD_MVN_OPTS_CURRENT has been stored with '-Dfull -Dquarkus.bootstrap.effective-model-builder' value<br/>[2022-05-27T12:55:40.293Z] ::notice title= [PRE] [Execution OK]::export BUILD_MVN_OPTS_CURRENT="-Dfull -Dquarkus.bootstrap.effective-model-builder"<br/>[2022-05-27T12:55:40.293Z] [INFO]  Execute command 'echo "BUILD_MVN_OPTS_CURRENT=-Dfull -Dquarkus.bootstrap.effective-model-builder"' in dir '/home/jenkins/workspace/KIE/kogito/main/pullrequest/optaplanner.tests.downstream.optaplanner-quickstarts'<br/>[2022-05-27T12:55:40.293Z] [command]/usr/bin/echo BUILD_MVN_OPTS_CURRENT=-Dfull -Dquarkus.bootstrap.effective-model-builder<br/>[2022-05-27T12:55:40.293Z] BUILD_MVN_OPTS_CURRENT=-Dfull -Dquarkus.bootstrap.effective-model-builder<br/>[2022-05-27T12:55:40.293Z] ::notice title= [PRE] [Execution OK]::echo "BUILD_MVN_OPTS_CURRENT=-Dfull -Dquarkus.bootstrap.effective-model-builder"<br/>[2022-05-27T12:55:40.293Z] ::endgroup::<br/>[2022-05-27T12:55:40.293Z] ::group::[Pull Request Flow] Execution Plan...<br/>[2022-05-27T12:55:40.293Z] [INFO]  ----------------------------------------------<br/>[2022-05-27T12:55:40.293Z] [INFO]  [3] projects will be executed<br/>[2022-05-27T12:55:40.293Z] <br/>[2022-05-27T12:55:40.293Z] [INFO]  [kiegroup/drools]<br/>[2022-05-27T12:55:40.293Z] [INFO]  Level Type: [upstream].<br/>[2022-05-27T12:55:40.293Z] [INFO]  mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }} ${{ env.DROOLS_BUILD_MVN_OPTS_UPSTREAM }}<br/>[2022-05-27T12:55:40.293Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-05-27T12:55:40.293Z] <br/>[2022-05-27T12:55:40.293Z] [INFO]  [kiegroup/optaplanner]<br/>[2022-05-27T12:55:40.293Z] [INFO]  Level Type: [upstream].<br/>[2022-05-27T12:55:40.293Z] [INFO]  mvn clean install -Dquickly ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_UPSTREAM }} ${{ env.OPTAPLANNER_BUILD_MVN_OPTS_UPSTREAM }}<br/>[2022-05-27T12:55:40.293Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-05-27T12:55:40.293Z] <br/>[2022-05-27T12:55:40.293Z] [INFO]  [kiegroup/optaplanner-quickstarts]<br/>[2022-05-27T12:55:40.293Z] [INFO]  Level Type: [current].<br/>[2022-05-27T12:55:40.293Z] [INFO]  mvn clean install -f optaplanner-quickstarts/pom.xml ${{ env.BUILD_MVN_OPTS }} ${{ env.BUILD_MVN_OPTS_CURRENT }} ${{ env.OPTAPLANNER_QUICKSTARTS_BUILD_MVN_OPTS }}<br/>[2022-05-27T12:55:40.293Z] [INFO]  [ [32m'docker system prune -f'[39m ]<br/>[2022-05-27T12:55:40.293Z] <br/>[2022-05-27T12:55:40.293Z] [INFO]  ----------------------------------------------<br/>[2022-05-27T12:55:40.293Z] ::endgroup::<br/>[2022-05-27T12:55:40.293Z] ::group::[Pull Request Flow] Checking out kiegroup/optaplanner and its dependencies<br/>[2022-05-27T12:55:40.293Z] [INFO]  Tree for project kiegroup/optaplanner-quickstarts. Dependencies: <br/>[2022-05-27T12:55:40.293Z] kiegroup/drools,<br/>[2022-05-27T12:55:40.293Z] kiegroup/optaplanner,<br/>[2022-05-27T12:55:40.293Z] kiegroup/optaplanner-quickstarts<br/>[2022-05-27T12:55:40.293Z] [INFO]  [kiegroup/drools] Checking out project<br/>[2022-05-27T12:55:40.556Z] [WARN]  project github.com/yurloc/drools:variable-listener-cleanup does not exist. It's not necessarily an error.<br/>[2022-05-27T12:55:40.812Z] [WARN]  project github.com/kiegroup/drools:variable-listener-cleanup does not exist. It's not necessarily an error.<br/>[2022-05-27T12:55:40.812Z] [INFO]  [kiegroup/drools] Checking out 'https://github.com/kiegroup/drools:main' into 'bc/kiegroup_drools'<br/>Superseded by #67<br/>[2022-05-27T12:55:44.342Z] Sending interrupt signal to process<br/>[2022-05-27T12:55:45.284Z] [ERROR]  command git clone --quiet --shallow-submodules --no-tags --branch main https://****@github.com/kiegroup/drools bc/kiegroup_drools failed with code 128. Error Message: error: git-remote-https died of signal 15<br/>[2022-05-27T12:55:45.284Z] fetch-pack: unexpected disconnect while reading sideband packet<br/>[2022-05-27T12:55:45.284Z] fatal: early EOF<br/>[2022-05-27T12:55:45.284Z] fatal: fetch-pack: invalid index-pack output<br/>[2022-05-27T12:55:45.284Z] <br/>[2022-05-27T12:55:45.288Z] script returned exit code 1<br/>[Pipeline] }<br/>[Pipeline] // withCredentials<br/>[Pipeline] }<br/>[2022-05-27T12:55:45.361Z] Deleting 1 temporary files<br/>[Pipeline] // configFileProvider<br/>[Pipeline] }<br/>[Pipeline] // script<br/>Post stage<br/>[Pipeline] junit<br/>[2022-05-27T12:55:45.476Z] Recording test results<br/>[2022-05-27T12:55:45.573Z] No test report files were found. Configuration error?<br/>Error when executing always post condition:<br/>Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to rhos-01-kie-psi-rhel7-xlarge-5<br/>		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1797)<br/>		at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:356)<br/>		at hudson.remoting.Channel.call(Channel.java:1001)<br/>		at hudson.FilePath.act(FilePath.java:1166)<br/>		at hudson.FilePath.act(FilePath.java:1155)<br/>		at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:107)<br/>		at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:153)<br/>		at hudson.tasks.junit.JUnitResultArchiver.parseAndSummarize(JUnitResultArchiver.java:247)<br/>		at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:63)<br/>		at hudson.tasks.junit.pipeline.JUnitResultsStepExecution.run(JUnitResultsStepExecution.java:29)<br/>		at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)<br/>		at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)<br/>hudson.AbortException: No test report files were found. Configuration error?<br/>	at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:154)<br/>	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3338)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:211)<br/>	at hudson.remoting.UserRequest.perform(UserRequest.java:54)<br/>	at hudson.remoting.Request$2.run(Request.java:376)<br/>	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)<br/>	at java.util.concurrent.FutureTask.run(FutureTask.java:266)<br/>	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)<br/>	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)<br/>	at java.lang.Thread.run(Thread.java:750)<br/><br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2022-05-27T12:55:45.954Z] + rm -rf console.log<br/>[Pipeline] sh<br/>[2022-05-27T12:55:46.517Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/optaplanner.tests.downstream.optaplanner-quickstarts/66/consoleText<br/>[2022-05-27T12:55:46.517Z] + tail -n 300<br/>
</details>

File: core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java [hunk 1]
```diff
@@ -385,7 +373,6 @@ public void afterEntityAdded(EntityDescriptor<Solution_> entityDescriptor, Objec
         if (lookUpEnabled) {
             lookUpManager.addWorkingObject(entity);
         }
-        variableListenerSupport.afterEntityAdded(entityDescriptor, entity);
```
Comment (triceo) [inline core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java:388]: I don't understand. How can we get away with not calling `varibleListenerSupport`? Who calls it instead?

**Review by triceo — COMMENTED** [review]
Overall a nice cleanup. I do have a question though.

**Review by triceo — APPROVED** [review]
LGTM then.

**Comment by triceo** [comment]
Jenkins retest this.

**Pull request merged**